src/phpDocumentor/Project/Version/DefinitionFactory.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * phpDocumentor
+ *
+ * PHP Version 5.4
+ *
+ * @copyright 2010-2014 Mike van Riel / Naenius (http://www.naenius.com)
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Project\Version;
+
+use phpDocumentor\DocumentGroupDefinitionFactory;
+use phpDocumentor\DocumentGuideFormat;
+use phpDocumentor\Project\DocumentGroup\Definition as DocumentGroupDefinition;
+use phpDocumentor\Project\VersionNumber;
+
+/**
+ * Factory for Version definition.
+ * Will use the registered factories to create the configured DocumentGroup\Definitions.
+ */
+class DefinitionFactory implements \phpDocumentor\DefinitionFactory
+{
+    /**
+     * @var DocumentGroupDefinitionFactory[]
+     */
+    private $documentGroupDefinitionFactories;
+
+    /**
+     * Creates a full provisioned version definition
+     *
+     * @param array $options
+     * @return Definition
+     */
+    public function create(array $options)
+    {
+        $documentGroups = $this->createDocumentGroupDefinitions($options);
+
+        return new Definition(
+            new VersionNumber($options['version']),
+            $documentGroups
+        );
+    }
+
+    /**
+     * Register a factory for later usage for a given type and format.
+     * Will override registered factories.
+     * The combination of type and format will identify a certain documentGroup
+     *
+     * @param string $type
+     * @param DocumentGuideFormat $format
+     * @param DocumentGroupDefinitionFactory $factory
+     */
+    public function registerDocumentGroupDefinitionFactory($type, DocumentGuideFormat $format, DocumentGroupDefinitionFactory $factory)
+    {
+        $this->documentGroupDefinitionFactories[$type][(string)$format] = $factory;
+    }
+
+    /**
+     * @param string $type
+     * @param string $format
+     * @return null|DocumentGroupDefinitionFactory
+     */
+    private function findFactory($type, $format)
```

Comment:
I recommend placing this method below createDocumentGroupDefinitions because it is only used there